### ISSUE TITLE
'no warnings' failure with perl 5.35.x

### ISSUE THREAD

**Issue by eserte** [sqitchers/sqitch#598]
t/add.t and t/rework.t fail with perl 5.35.6 if Template.pm is installed:
```
#   Failed test 'no warnings'
#   at /opt/perl-5.35.6/lib/5.35.6/Test/Builder.pm line 193.
# There were 4 warning(s)
#     Previous test 75 'The template should have been evaluated with requires and conflicts'
#     each on anonymous hash will always start from the beginning at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75.
#  at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75.
# 	require Template/Document.pm called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Provider.pm line 46
# 	Template::Provider::BEGIN() called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval {...} called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	require Template/Provider.pm called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template.pm line 29
# 	Template::BEGIN() called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval {...} called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	require Template.pm called at (eval 347) line 1
# 	main::BEGIN() called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval {...} called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval 'use Template; 1' called at t/add.t line 462
... etc ...
# Looks like you failed 1 test of 236.
t/add.t ............. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/236 subtests 
...
#   Failed test 'no warnings'
#   at /opt/perl-5.35.6/lib/5.35.6/Test/Builder.pm line 193.
# There were 4 warning(s)
#     Previous test 23 'test-rework/verify/foo.sql does not exist'
#     each on anonymous hash will always start from the beginning at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75.
#  at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75.
# 	require Template/Document.pm called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Provider.pm line 46
# 	Template::Provider::BEGIN() called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval {...} called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	require Template/Provider.pm called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template.pm line 29
# 	Template::BEGIN() called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval {...} called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	require Template.pm called at (eval 554) line 1
# 	App::Sqitch::Command::add::BEGIN() called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval {...} called at /home/cpansand/.local/share/.cpan/build/2021112114/Template-Toolkit-3.009-0/blib/lib/Template/Document.pm line 75
# 	eval 'use Template; 1' called at /home/cpansand/.cpan/build/2021112114/App-Sqitch-v1.2.0-2/blib/lib/App/Sqitch/Command/add.pm line 401
# 	App::Sqitch::Command::add::_add(App::Sqitch::Command::add=HASH(0x55eca0b540c8), "foo", Path::Class::File=HASH(0x55eca1a89858), Path::Class::File=HASH(0x55eca11cbff0), "sqlite", "empty") called at /home/cpansand/.cpan/build/2021112114/App-Sqitch-v1.2.0-2/blib/lib/App/Sqitch/Command/add.pm line 342
# 	App::Sqitch::Command::add::execute(undef) called at t/rework.t line 159
...
# Looks like you failed 1 test of 234.
t/rework.t .......... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/234 subtests 
```

It's very likely that this is related to https://github.com/abw/Template2/issues/287
So there's not much you can do about it then waiting for a fix in Template-Toolkit --- or maybe, if the fail reports are too annoying, disable Test::NoWarnings in the failing test files for perl >= 5.35.2 and Template <= 3.009.

**Comment by theory** [comment]
I'm fine to wait to see if it gets fixed for now, since 5.35 is pre-release.

**Comment by eserte** [comment]
The Template-Toolkit issue looks like it is resolved. Closing this ticket.
